src/context/actions.ts

```diff
@@ -0,0 +1,115 @@
+import { ParsedUrlQuery } from "querystring";
+import { Dispatch } from "react";
+import { LOGIN_PROVIDERS_URL, LOGOUT_URL, USER_INFO_URL } from "../env";
+import { UserStateAction } from "../types/Login";
+
+/**
+ * Performs the callback POST request to check 3rd partyauthentication
+ * was successful. Fetches user data on success.
+ * @param dispatch Reducer dispatch function used to update user context
+ * @param error Function for displaying errors (usually component state)
+ * @param query URL query object with code and state to POST to backend
+ */
+export async function login(
+  dispatch: Dispatch<UserStateAction>,
+  error: (msg: string) => void,
+  query: ParsedUrlQuery
+) {
+  dispatch({type: 'loading'})
+
+  let res: Response
+  try {
+    res = await fetch(`${LOGIN_PROVIDERS_URL}/github`, {
```

Comment:
What happens when we have other login providers?  This feels quite static to github.  Do we need a TODO to make this more generic when we add a second provider?

Reply:
Yeah :+1: Thinking for future providers is to use dynamic routing (the pages like `[query]`) and can then more generically handle the auth redirect callbacks. Reason I just went with static for now is because unsure if the required behaviour for future providers will be different or if it will always be this `status` and `code` post requesting.

Will add a todo to this effect.

Reply:
Any oauth (or openid-connect) style provider will be very similar if not identical in flow.  It'll only really differ for other kinds of providers.